doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,115 +1257,87 @@ Managing the local context
    for any unresolved variables rather than failing.  Typically this happens when
    using a ``%`` intropattern (see :n:`@simple_intropattern`).
 
-.. tacn:: clear @ident
-   :name: clear
+.. tacn:: clear {? {? - } {+ @ident } }
 
-   This tactic erases the hypothesis named :n:`@ident` in the local context of
-   the current goal. As a consequence, :n:`@ident` is no more displayed and no
-   more usable in the proof development.
+   clear
+   clear ident ...
+   clear - ident ...
 
-   .. exn:: No such hypothesis.
-      :undocumented:
+   Erases hypotheses from the context of the current goal, provided that the
+   remaining hypotheses and the goal don't depend directly or indirectly on the
+   erased hypotheses.
+   That means the hypotheses will no longer appear in the context and therefore
+   can't be used in subsequent proof steps.
+
+   :n:`clear`
+     All hypotheses are erased.  This may leave the context unchanged; this form never fails.
```

Comment:
```suggestion
     All hypotheses that can be erased are erased.  This may leave the context unchanged; this form never fails.
```

Or:

```suggestion
     All hypotheses are erased except those that the goal depends on.  This may leave the context unchanged; this form never fails.
```

Reply:
The paragraph just above says "Erases hypotheses ... provided that the remaining hypotheses and the goal don't depend directly or indirectly on the erased hypotheses."  I didn't want to repeat that wordy statement in each of the items and I couldn't think of something more succinct that could be repeated.  "All erasable hypotheses are erased"?  "All hypotheses will be erased if possible"?

Reply:
"All erasable hypotheses are erased" sounds good to me.

Reply:
Ah, it sounds like a tautology.